examples/react-vanilla-ts-src/src/server.ts

```diff
@@ -0,0 +1,37 @@
+
+import Fastify from 'fastify'
+import FastifyVite from '@fastify/vite'
+import { renderToString } from 'react-dom/server'
+import { resolve } from 'path'
+
+export async function main (dev?: boolean) {
+  const server = Fastify()
+
+  await server.register(FastifyVite, {
+    // The compiled server will live in <root>/build
+    // which is the same depth as <root>/src, so 
+    // we can use import.meta.dirname here
+    root: resolve(import.meta.dirname, '..'),
+    distDir: resolve(import.meta.dirname, '..', 'build'),
```

Comment:
Did you want to add a comment here saying it must match the vite.config.js's `build.outDir` folder?

Reply:
Good idea :)